lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -33,20 +70,183 @@ def onCreate(self, *args):
+
+        # Get all the other Metadata and remove the ones above, as well as a few
+        # we don't ever want to show.
+        allMetadata = self.prim.GetAllMetadata()
+        keysToDelete = ['kind', 'active', 'instanceable', 'typeName', 'documentation']
+        for key in keysToDelete:
+            allMetadata.pop(key, None)
+        if allMetadata:
+            cmds.separator(h=10, style='single', hr=True)
+
+            for k in allMetadata:
+                # All extra metadata is for display purposes only - it is not editable, but we
+                # allow keyboard focus so you copy the value.
+                mdLabel = getPrettyName(k) if useNiceName else k
+                self.extraMetadata[k] = cmds.textFieldGrp(label=mdLabel, editable=False, enableKeyboardFocus=True)
+
+        # Update all metadata values.
+        self.refresh()
+
+    def refresh(self):
+        # PrimPath
+        cmds.textFieldGrp(self.primPath, edit=True, text=str(self.prim.GetPath()))
+
+        # Kind
+        model = Usd.ModelAPI(self.prim)
+        primKind = model.GetKind()
+        if not primKind:
+            # Special case to handle the empty string (for meta data value empty).
+            cmds.optionMenuGrp(self.kind, edit=True, select=1)
+        else:
+            cmds.optionMenuGrp(self.kind, edit=True, value=primKind)
+
+        # Active
+        cmds.checkBoxGrp(self.active, edit=True, value1=self.prim.IsActive())
+
+        # Instanceable
+        cmds.checkBoxGrp(self.instan, edit=True, value1=self.prim.IsInstanceable())
+
+        # All other metadata types
+        for k in self.extraMetadata:
+            v = self.prim.GetMetadata(k) if k != 'customData' else self.prim.GetCustomData()
+            cmds.textFieldGrp(self.extraMetadata[k], edit=True, text=str(v))
+
+    def _onKindChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            model = Usd.ModelAPI(self.prim)
+            model.SetKind(value)
+
+    def _onActiveChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            self.prim.SetActive(value)
+
+    def _onInstanceableChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            self.prim.SetInstanceable(value)
+
+class NoticeListener(object):
+    # Inserted as a custom control, but does not have any UI. Instead we use
+    # this control to be notified from USD when any metadata has changed
+    # so we can update the AE fields.
+    def __init__(self, prim, metadataControls):
+        self.prim = prim
+        self.metadataControls = metadataControls
+
+    def onCreate(self, *args):
+        self.listener = Tf.Notice.Register(Usd.Notice.ObjectsChanged,
+                                           self.__OnPrimsChanged, self.prim.GetStage())
+        pname = cmds.setParent(q=True)
+        cmds.scriptJob(uiDeleted=[pname, self.onClose], runOnce=True)
```

Comment:
The metadata section is collapsed by default. This method is only called when it is expanded. So we won't unnecessarily have the usd notice.